go/app/main.go

```diff
@@ -15,27 +19,145 @@ import (
 const (
 	ImgDir = "images"
 )
-
 type Response struct {
-	Message string `json:"message"`
+	Message    string `json:"message"`
 }
 
+// Define structure
+type ItemIndex struct {
+	Items      []Item `json:"items"`
+}
+type Item struct {	
+	Name       string `json:"name"`
+	Category   string `json:"category"`
+	Image_name string `json:"image_name"`
+}
+
+// GET "/"
 func root(c echo.Context) error {
 	res := Response{Message: "Hello, world!"}
 	return c.JSON(http.StatusOK, res)
 }
 
+var itemindex ItemIndex
+var item Item
+
+// GET "/items"
+func getItem(c echo.Context) error {
+	//open JSON file
+	file, err := os.Open("items.json")
+	if err != nil {
+		c.Logger().Infof("Error message: %s", err)
```

Comment:
ここでerrorが出た場合に、続きの処理をするとずっとエラーが出るのでエラーが出たらエラーのreturnするようにしましょう。ログだけだとそのまま処理が続いてしまいます。

```suggestion
        return c.JSON(http.StatusInternalServerError, Response{Message: err.Error()})
```
こんな感じです。

要はerrorの型を返すようにしてDebugできるようにしたいです。